File: dixit-companion/src/main/java/fr/erban/dxitcompanion/db/player/PlayerViewModel.java
package fr.erban.dxitcompanion.db.player;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

public class PlayerViewModel extends AndroidViewModel {

    private final LiveData<List<PlayerEntity>> players;
    private final PlayerRepository playerRepository;

    public PlayerViewModel(Application application) {
        super(application);
        playerRepository = new PlayerRepository(application);
        players = playerRepository.getAllPlayers();
    }

    public LiveData<List<PlayerEntity>> getPlayers() {
        return players;
    }

    public void insert(PlayerEntity player) {
        playerRepository.insert(player);
    }

    public void update(List<PlayerEntity> playersToUpdate) {
        playerRepository.update(playersToUpdate);
    }
}

File: dixit-companion/src/main/java/fr/erban/dxitcompanion/db/player/PlayerViewModel.kt
package fr.erban.dxitcompanion.db.player

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.viewModelScope
import kotlinx.coroutines.launch

class PlayerViewModel(application: Application) : AndroidViewModel(application) {

    private val playerRepository = PlayerRepository(application)
    val players: LiveData<List<PlayerEntity>> = playerRepository.allPlayers

    fun insert(player: PlayerEntity) {
        viewModelScope.launch {
            playerRepository.insert(player)
        }
    }

    fun update(players: List<PlayerEntity>) {
        viewModelScope.launch {
            playerRepository.update(players)
        }
    }
}
